lib/bolt/catalog.rb

```diff
@@ -69,40 +95,7 @@ def compile_catalog(request)
           Puppet.override(bolt_pdb_client: pdb_client,
                           bolt_inventory: inv) do
             Puppet.lookup(:pal_current_node).trusted_data = target['trusted']
-            pal.with_catalog_compiler do |compiler|
-              # Deserializing needs to happen inside the catalog compiler so
-              # loaders are initialized for loading
-              plan_vars = Puppet::Pops::Serialization::FromDataConverter.convert(request['plan_vars'])
```

Comment:
This is the line of code that makes `Target`, `Result`, etc objects work inside the apply block.

Reply:
Just passing them into `with_catalog_compiler` doesn't cause them to be rehydrated into Puppet types. They're just passed along as regular hashes.

Reply:
Oh. I thought the spec tests were passing. I was looking at the wrong cell